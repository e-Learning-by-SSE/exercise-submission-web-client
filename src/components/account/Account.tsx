import React, { Component } from 'react'
import { UserDto } from 'stumgmtbackend';
import { Dropdown, Icon } from 'semantic-ui-react';
import Help from './Help';


export default class Account extends Component<React.PropsWithChildren<{user: UserDto}>, {dropdown: any, selectedItem: any}> {

    constructor(props: React.PropsWithChildren<{user: UserDto}>) {
        super(props);
        this.state = {dropdown: this.createAccountDropdown(),
            selectedItem: null};
      }
      
      handleComboBoxSelection = (event: React.SyntheticEvent<HTMLElement, Event>, data: any) => {
        event.preventDefault();
        let id = data.value;
        switch(id) {
          case "help":
            this.setState({selectedItem: <Help onClosed={this.onMenuClosed}/>});
            break;

        }
       
      }

    onMenuClosed = () => {
      this.setState({selectedItem: null})
    }


    private createAccountDropdown() {
      const trigger = (
        <span>
          <Icon name='user' /> Hello, {this.props.user.username}
        </span>
      )

      
      const options = [
        {
          key: 'user',
          text: (
            <span>
              Signed in as <strong>{this.props.user.username}</strong>
            </span>
          ),
          disabled: true,
        },
        { key: 'help', value:'help', text: 'Help' },
        { key: 'settings', value:'settings', text: 'Settings' },
        { key: 'sign-out', value:'sign-out', text: 'Sign Out' },
      ]

      return <Dropdown trigger={trigger} options={options} onChange={this.handleComboBoxSelection} />;
    }


    render() {
        return (
           <div className="accountDropdown">
               {this.state.dropdown}
               {this.state.selectedItem}
           </div>
        )
    }
}



