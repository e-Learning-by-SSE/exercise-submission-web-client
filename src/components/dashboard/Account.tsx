import React, { Component } from 'react'
import { UserDto } from 'stumgmtbackend';
import { Dropdown, Icon } from 'semantic-ui-react';


export default class Account extends Component<React.PropsWithChildren<{user: UserDto}>, {dropdown: any}> {

    constructor(props: React.PropsWithChildren<{user: UserDto}>) {
        super(props);
        this.state = {dropdown: this.createAccountDropdown()};
    }

    private createAccountDropdown() {
      const trigger = (
        <span>
          <Icon name='user' /> Hello, {this.props.user.username}
        </span>
      )
      
      const options = [
        {
          key: 'user',
          text: (
            <span>
              Signed in as <strong>{this.props.user.username}</strong>
            </span>
          ),
          disabled: true,
        },
        { key: 'help', text: 'Help' },
        { key: 'settings', text: 'Settings' },
        { key: 'sign-out', text: 'Sign Out' },
      ]

      return <Dropdown trigger={trigger} options={options} />;
    }


    render() {
        return (
           <div className="accountDropdown">
               {this.state.dropdown}
           </div>
        )
    }
}



