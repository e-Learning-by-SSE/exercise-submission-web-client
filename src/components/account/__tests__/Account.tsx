import renderer from "react-test-renderer";
import Account from "../Account";
import { UserDto, UserDtoRoleEnum } from "stumgmtbackend";

it("render Account",()   => {
    const userDto: UserDto = {
        username: "test",
        id: "1",
        role: UserDtoRoleEnum.USER,
        displayName: "test"
    }
    const account = renderer.create(<Account user={userDto}></Account>).toJSON();
    expect(account).toMatchSnapshot();
});
