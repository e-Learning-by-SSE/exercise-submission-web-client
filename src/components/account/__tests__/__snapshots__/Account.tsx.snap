// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Account 1`] = `
<div
  className="accountDropdown"
>
  <div
    aria-expanded={false}
    className="ui dropdown"
    onBlur={[Function]}
    onChange={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="listbox"
    tabIndex={0}
  >
    <span>
      <i
        aria-hidden="true"
        className="user icon"
        onClick={[Function]}
      />
       Hello, 
      test
    </span>
    <i
      aria-hidden="true"
      className="dropdown icon"
      onClick={[Function]}
    />
    <div
      className="menu transition"
    >
      <div
        aria-checked={false}
        aria-disabled={true}
        aria-selected={false}
        className="disabled item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Signed in as 
          <strong>
            test
          </strong>
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={true}
        className="selected item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Help
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Settings
        </span>
      </div>
      <div
        aria-checked={false}
        aria-selected={false}
        className="item"
        onClick={[Function]}
        role="option"
        style={
          Object {
            "pointerEvents": "all",
          }
        }
      >
        <span
          className="text"
        >
          Sign Out
        </span>
      </div>
    </div>
  </div>
</div>
`;
